### ISSUE TITLE
Flust timestamp are not aligned

### ISSUE THREAD

**Issue by luca3m** [armon/statsite#206]
When configuring statsite with a flush time of 10s I noticed that the flush timestamp it's not aligned. 

I get these results (the result depends on the timestamp of statsite launch):

```
counts.test|4.000000|1468836448
counts.test|4.000000|1468836458
counts.test|4.000000|1468836468
```

And not

```
counts.test|4.000000|1468836440
counts.test2|4.000000|1468836450
counts.test4|4.000000|1468836460
```

Would be better the latter, wouldn't it?

**Comment by luca3m** [comment]
I've been able to make it work using `ev_periodic` instead of `ev_timer`, see [manpage](http://linux.die.net/man/3/ev). I think it's important for a tool like this to use absolute timer instead of relative ones.

Looks like `ae` has not a native support for this kind of events, statsite should calculate by itself the drift between two flush events.

**Comment by armon** [comment]
@luca3m Do you mean `ae` is causing drift? Or that it is just not aligned with the 10 second interval?

**Comment by luca3m** [comment]
The latter. It has never been aligned. With `libev` is easier to implement it, on `ae` it's not. If you are interested on merging this I can try to provide a patch.

**Comment by armon** [comment]
@luca3m Yeah, I think this should be configurable behavior, but it would be nice to have an strict alignment mode!

**Comment by dankuna** [comment]
One other feature idea to perhaps tack on with the strict alignment mode would be a random offset between the end of the flush interval and when the data is dumped to the sink. If you're using a sink that then forwards to a third-party logging service it's not necessarily desirable to have all clients pushing at exactly the same time. Granted, this could be added in to the sink itself, but might be better off as a universal feature.
